ner_v2/detectors/temporal/date/date_detection.py

```diff
@@ -0,0 +1,706 @@
+                      'detection_method': model_not_verified
+
+                    }
+                 ]
+
+        """
+        date_dict_list = []
+        model_object = Models()
+        model_output = model_object.run_model(entity_type=model_constant.DATE_ENTITY_TYPE,
+                                              bot_message=self.bot_message, user_message=self.text)
+        for output in model_output:
+            entity_value_list, original_text_list = self._date_value(text=output[model_constant.MODEL_DATE_VALUE])
+            if entity_value_list:
+                date_value = entity_value_list[0]
+                detection_method = FROM_MODEL_VERIFIED
+            else:
+                date_value = output[model_constant.MODEL_DATE_VALUE]
+                detection_method = FROM_MODEL_NOT_VERIFIED
+
+            date_dict_list.append(
+                {
+                    detector_constant.DATE_VALUE: date_value,
+                    detector_constant.ORIGINAL_DATE_TEXT: output[model_constant.MODEL_DATE_VALUE],
+                    detector_constant.DATE_FROM_PROPERTY: output[model_constant.MODEL_DATE_FROM],
+                    detector_constant.DATE_TO_PROPERTY: output[model_constant.MODEL_DATE_TO],
+                    detector_constant.DATE_START_RANGE_PROPERTY: output[model_constant.MODEL_START_DATE_RANGE],
+                    detector_constant.DATE_END_RANGE_PROPERTY: output[model_constant.MODEL_END_DATE_RANGE],
+                    detector_constant.DATE_NORMAL_PROPERTY: output[model_constant.MODEL_DATE_NORMAL],
+                    detector_constant.DATE_DETECTION_METHOD: detection_method
+                }
+
+            )
+        return date_dict_list
+
+
+class DateDetector(BaseDetector):
+    """
+    Detects date in various formats from given text and tags them.
+
+    Detects all date entities in given text and replaces them by entity_name.
+    Additionally detect_entity() method returns detected date values in dictionary containing values for day (dd),
+    month (mm), year (yy), type (exact, possible, everyday, weekend, range etc.)
+
+    Attributes:
+        text: string to extract entities from
+        entity_name: string by which the detected date entities would be replaced with on calling detect_entity()
+        tagged_text: string with date entities replaced with tag defined by entity name
+        processed_text: string with detected date entities removed
+        date: list of date entities detected
+        original_date_text: list to store substrings of the text detected as date entities
+        tag: entity_name prepended and appended with '__'
+        timezone: Optional, pytz.timezone object used for getting current time, default is pytz.timezone('UTC')
+        now_date: datetime object holding timestamp while DateDetector instantiation
+        bot_message: str, set as the outgoing bot text/message
+        language: source language of text
+    """
+
+    def supported_languages(self):
```

Comment:
This needs to be populated based on language folders which are present.

Reply:
done